tests/test_profile_stability.py

```diff
@@ -0,0 +1,145 @@
+                filename = os.path.join(root, basename)
+                found.append(filename)
+    return found
+
+
+def corresponding_product_built(build_dir, reference_fname):
+    ref_path_components = reference_fname.split(os.path.sep)
+    product_id = ref_path_components[-2]
+    return os.path.isdir(os.path.join(build_dir, product_id))
+
+
+def get_matching_compiled_profile_filename(build_dir, reference_fname):
+    ref_path_components = reference_fname.split(os.path.sep)
+    product_id = ref_path_components[-2]
+    profile_fname = ref_path_components[-1]
+    matching_filename = os.path.join(build_dir, product_id, "profiles", profile_fname)
+    if os.path.isfile(matching_filename):
+        return matching_filename
+
+
+def get_selections_key_from_yaml(yaml_fname):
+    return ssg.yaml.open_raw(yaml_fname)["selections"]
+
+
+def get_profile_name_from_reference_filename(fname):
+    path_components = fname.split(os.path.sep)
+    product_id = path_components[-2]
+    profile_id = os.path.splitext(path_components[-1])[0]
+    name = "{product_id}'s {profile_id}".format(
+        product_id=product_id, profile_id=profile_id)
+    return name
+
+
+def get_reference_vs_built_difference(reference_fname, built_fname):
+    ref_selections = get_selections_key_from_yaml(reference_fname)
+    built_selections = get_selections_key_from_yaml(built_fname)
+    difference = compare_sets(ref_selections, built_selections)
+    return difference
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("build_root")
+    parser.add_argument("test_data_root")
+    args = parser.parse_args()
+
+    reference_files = get_references(args.test_data_root)
+    if not reference_files:
+        raise RuntimeError("Unable to find any reference profiles in {test_root}"
+                           .format(test_root=args.test_data_root))
+    no_changes = True
+    for ref in reference_files:
+        if not corresponding_product_built(args.build_root, ref):
+            continue
+
+        compiled_profile = get_matching_compiled_profile_filename(args.build_root, ref)
+        if not compiled_profile:
+            msg = ("Unexpectedly unable to find compiled profile corresponding"
+                   "to the test file {ref}, although the corresponding product has been built. "
+                   "This indicates that a profile we have tests for is missing."
+                   .format(ref=ref))
+            raise RuntimeError(msg)
+        difference = get_reference_vs_built_difference(ref, compiled_profile)
+        if not difference.empty:
+            no_changes = False
+            comprehensive_profile_name = get_profile_name_from_reference_filename(ref)
+            report_comparison(comprehensive_profile_name, difference)
+
+    if not no_changes:
+        msg = (
+            "If changes to mentioned profiles are intentional, "
+            "edit the appropriate files in '{test_root}' directory."
```

Comment:
It may be worthy to print out a `cp` command to make it easy for the user to align the profile with the reference file.